src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -201,6 +201,51 @@ def __init__(self, target: Text,
 
         self._channel = cygrpc.AioChannel(_common.encode(target))
 
+    def get_state(self,
+                  try_to_connect: bool = False) -> grpc.ChannelConnectivity:
+        """Check the connectivity state of a channel.
+
+        This is an EXPERIMENTAL API.
+
+        If the channel reaches a stable connectivity state, it is guaranteed
+        that the return value of this function will eventually converge to that
+        state.
+
+        Args: try_to_connect: a bool indicate whether the Channel should try to
+          connect to peer or not.
+
+        Returns: A ChannelConnectivity object.
+        """
+        result = self._channel.check_connectivity_state(try_to_connect)
+        return _common.CYGRPC_CONNECTIVITY_STATE_TO_CHANNEL_CONNECTIVITY[result]
+
+    async def wait_for_state_change(
+            self,
+            last_observed_state: grpc.ChannelConnectivity,
+    ) -> None:
+        """Wait for a change in connectivity state.
+
+        This is an EXPERIMENTAL API.
+
+        The function blocks until there is a change in the channel connectivity
+        state from the "last_observed_state". If the state is already
+        different, this function will return immediately.
+
+        There is an inherent race between the invocation of
+        "Channel.wait_for_state_change" and "Channel.get_state". The state can
+        change arbitrary times during the race, so there is no way to observe
+        every state transition.
+
+        If there is a need to put a timeout for this function, please refer to
+        "asyncio.wait_for".
+
+        Args:
+          last_observed_state: A grpc.ChannelConnectivity object representing
+            the last known state.
+        """
+        assert await self._channel.watch_connectivity_state(
```

Comment:
`watch_connectivity_state` returns `True` when it observed a state change, and it returns `False` when it times out . I was wrong in the comment that a channel close will trigger a failure, because before channel shuts down, the connectivity state will switch into `SHUTDOWN`, so it will clean all pending `watch_connectivity_state` to return `True`.

In short, for this design, the `watch_connectivity_state` should only return `True`, otherwise it suggests problems in our implementation.

I updated the `test_shutdown` case to better ensure if the Core changes its behavior, and the test will break.